plugin/trino-spanner/src/main/java/io/trino/plugin/spanner/SpannerTableProperties.java

```diff
@@ -0,0 +1,110 @@
+package io.trino.plugin.spanner;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.plugin.jdbc.TablePropertiesProvider;
+import io.trino.spi.session.PropertyMetadata;
+import io.trino.spi.type.ArrayType;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static io.trino.spi.session.PropertyMetadata.booleanProperty;
+import static io.trino.spi.session.PropertyMetadata.stringProperty;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class SpannerTableProperties
+        implements TablePropertiesProvider
+{
+    public static final String PRIMARY_KEYS = "primary_keys";
+    public static final String NOT_NULL_FIELDS = "not_null_fields";
+    public static final String COMMIT_TIMESTAMP_FIELDS = "commit_timestamp_fields";
+    public static final String INTERLEAVE_IN_PARENT = "interleave_in_parent";
+    public static final String ON_DELETE_CASCADE = "on_delete_cascade";
```

Comment:
We want to minimize the scope when adding a new connector. Please remove these properties as much as possible. I suppose only `primary_keys` is required in this PR.

Reply:
@ebyhr Can we keep interleave_in_parent as well? That is the base of how spanner works and I think it should be considered for this PR as implementing that is very easy.

Reply:
Without interleaved tables all joins when pushed down to spanner would make it costly to process